mava/evaluator.py

```diff
@@ -93,7 +93,12 @@ def not_done(carry: Tuple) -> bool:
         }
         # Log won episode if win rate is required.
         if log_win_rate:
-            eval_metrics["won_episode"] = jnp.all(final_state.timestep.reward >= 1.0).astype(int)
+            if "won_episode" not in final_state.timestep.extras:
+                eval_metrics["won_episode"] = jnp.all(final_state.timestep.reward >= 1.0).astype(
+                    int
+                )
+            else:
+                eval_metrics["won_episode"] = final_state.timestep.extras["won_episode"]
```

Comment:
This suggestion for better readability
:
```suggestion
            if "won_episode" not in final_state.timestep.extras:
                eval_metrics["won_episode"] = final_state.timestep.extras["won_episode"]
            else:
                eval_metrics["won_episode"] = int(jnp.all(final_state.timestep.reward >= 1.0))
```